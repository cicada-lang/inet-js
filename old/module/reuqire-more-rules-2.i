import Nat, zero, add1 from "./reuqire-more-rules-1.i"

node add
  Nat :target!
  Nat :addend
  --------
  Nat :result
end

rule zero add
  (add)-addend
  result-(add)
end

rule add1 add
  (add)-addend
  (add1)-prev add
  add1 result-(add)
end
