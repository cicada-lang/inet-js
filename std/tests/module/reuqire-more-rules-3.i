import { zero, add1, add } from "./reuqire-more-rules-2.i"

eval @inspect(@run(add(zero(), zero())))
eval @inspect(@run(add(add1(zero()), add1(zero()))))
eval @inspect(@run(add(add1(add1(zero())), add1(add1(zero())))))
