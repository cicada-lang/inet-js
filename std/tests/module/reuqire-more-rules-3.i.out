netFromPort (zero₁)-value! end
netFromPort (add1₂)-value!
  (add1₂)-prev value-(add1₁)
  (add1₁)-prev value-(zero₃)
end
netFromPort (add1₇)-value!
  (add1₇)-prev value-(add1₈)
  (add1₈)-prev value-(add1₅)
  (add1₅)-prev value-(add1₆)
  (add1₆)-prev value-(zero₅)
end
